File: tests/foreachloop/foreachloop.cs
using System;
using System.Diagnostics;
using System.IO;

public class foreachloop
{
	public static void Main()
	{
	    Stopwatch timer = new Stopwatch();
        timer.Start();
        foreachlooptest();
		timer.Stop();
		Console.Write(timer.ElapsedMilliseconds.ToString());
	}
	
	static void foreachlooptest()
	{
        string[] heavycars = new string[] { "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Volvo", "BMW", "Ford", "Mazda", "Mazda", "Volvo" };
        foreach (string car1 in heavycars)
        {
	    	foreach (string car2 in heavycars)
            {
	    	    foreach (string car3 in heavycars)
                {

                }
            }
        }
	}		
}

File: tests/foreachloop/foreachloop.vb
Imports System
Imports System.Diagnostics
Imports System.IO

Public Class forloopeach
    Public Shared Sub Main()
        Dim timer As Stopwatch = New Stopwatch()
        timer.Start()
        forloopeachtest()
        timer.Stop()
        Console.WriteLine(timer.ElapsedMilliseconds.ToString())
    End Sub

    Private Shared Sub forloopeachtest()
        Dim heavycars As String() = New String() {"Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda","Volvo", "BMW", "Ford", "Mazda", "Mazda","Volvo"} 
        For each car As String In heavycars
		
        Next
        For each car As String In heavycars
		
        Next
        For each car As String In heavycars
		
        Next
        For each car As String In heavycars
		
        Next
        For each car As String In heavycars
		
        Next
    End Sub
End Class
